File: src/main/java/de/lulonaut/Bot/utils/Cache.java
package de.lulonaut.Bot.utils;

import java.util.Map;

public class Cache {

    public static Map<String, Map<String, String>> cache;

    public static void refreshOrAddCache(String GuildID) {
        cache.put(GuildID, Database.loadConfig(GuildID));
    }

    public static Map<String, String> getConfig(String GuildID) {
        if (!cache.containsKey(GuildID)) {
            refreshOrAddCache(GuildID);
        }
        return cache.get(GuildID);
    }
}

File: src/main/kotlin/de/lulonaut/bot/utils/Cache.kt
package de.lulonaut.bot.utils

object Cache {
    private var cache: MutableMap<String, Map<String?, String?>?>? = HashMap()
    fun refreshOrAddCache(GuildID: String) {
        println("database request!")
        cache!![GuildID] = Database.loadConfig(GuildID)
    }

    fun getConfig(GuildID: String): Map<String?, String?>? {
        if (!cache!!.containsKey(GuildID)) {
            refreshOrAddCache(GuildID)
        }
        return cache!![GuildID]
    }
}